Diagnostics/FreeBusyChecker/FreeBusyChecker.ps1

```diff
@@ -0,0 +1,326 @@
+            AuthConfigCheck
+            PrintDynamicWidthLine
+            CurrentCertificateThumbprintCheck
+            PrintDynamicWidthLine
+            AutoDVirtualDCheckOAuth
+            $AutoDiscoveryVirtualDirectoryOAuth
+            PrintDynamicWidthLine
+            EWSVirtualDirectoryCheckOAuth
+            PrintDynamicWidthLine
+            AvailabilityAddressSpaceCheckOAuth
+            PrintDynamicWidthLine
+            OAuthConnectivityCheck
+            PrintDynamicWidthLine
+        }
+        #endregion
+    }
+    # EXO Part
+    if ($Org -contains 'ExchangeOnline' -OR -not $Org) {
+        #region ConnectExo
+
+        Write-Host -ForegroundColor Green $CollectingExoAvailabilityInformation
+        # Check if the ExchangeOnlineManagement module is already installed
+        if (-not (Get-Module -Name ExchangeOnlineManagement -ListAvailable)) {
+            # If not installed, then install the module
+            Write-Host -ForegroundColor Yellow $ExchangeOnlinePowershellModuleMessage
+            Install-Module -Name ExchangeOnlineManagement -Force
+            PrintDynamicWidthLine
+        } else {
+            Write-Host $ExchangeOnlineModuleAvailableMessage
+            $ExoModuleVersion = Get-Module -Name ExchangeOnlineManagement -ListAvailable | Format-List name, Version
+            $ExoModuleVersion
+            PrintDynamicWidthLine
+        }
+
+        Connect-ExchangeOnline -ShowBanner:$false
+
+        $Script:ExoOrgRel = Get-OrganizationRelationship | Where-Object { ($_.DomainNames -like $ExchangeOnPremDomain ) } | Select-Object Enabled, Identity, DomainNames, FreeBusy*, TarGet*
+        $ExoIntraOrgCon = Get-IntraOrganizationConnector | Select-Object Name, TarGetAddressDomains, DiscoveryEndpoint, Enabled
+        $tarGetAddressPr1 = ("https://AutoDiscover." + $ExchangeOnPremDomain + "/AutoDiscover/AutoDiscover.svc/WSSecurity")
+        $tarGetAddressPr2 = ("https://" + $ExchangeOnPremDomain + "/AutoDiscover/AutoDiscover.svc/WSSecurity")
+        exoHeaderHtml
+
+        #endregion
+
+        #region ExoDAuthCheck
+        if ($Auth -like "DAuth" -or -not $Auth -or $Auth -like "All") {
+            PrintDynamicWidthLine
+            Write-Host $TestingExoDAuthConfiguration
+
+            ExoOrgRelCheck
+            PrintDynamicWidthLine
+            EXOFedOrgIdCheck
+            PrintDynamicWidthLine
+            ExoTestOrgRelCheck
+            SharingPolicyCheck
+        }
+        #endregion
+
+        #region ExoOauthCheck
+        if ($Auth -like "OAuth" -or -not $Auth -or $Auth -like "All") {
+            Write-Host $TestingExoOAuthConfiguration
+
+            ExoIntraOrgConCheck
+            PrintDynamicWidthLine
+            EXOIntraOrgConfigCheck
+            PrintDynamicWidthLine
+            EXOAuthServerCheck
+            PrintDynamicWidthLine
+            ExoTestOAuthCheck
+            PrintDynamicWidthLine
+        }
+        #endregion
+
+        Disconnect-ExchangeOnline  -Confirm:$False
```

Comment:
We should not be disconnecting from Exchange Online powershell session. We don't want to be modifying admins current connections.

Reply:
Removing